From 263436931d36368746f661fb10afcf808badbcf2 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Thu, 3 Nov 2011 23:27:18 +0800
Subject: [PATCH 168/249] ENGR00161320 vpu: workaround to disable BWB flag on
 i.mx6q

workaround to disable BWB flag on i.mx6q since it may cause vpu
block with current v2.1.2 f/w.

Will remove the patch after f/w is upgraded.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 356f43b..fc24543 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -211,10 +211,7 @@ RetCode vpu_Init(void *cb)
 		    BUF_PIC_FLUSH << BIT_BUF_PIC_FLUSH | BUF_PIC_RESET <<
 		    BIT_BUF_PIC_RESET;
 		VpuWriteReg(BIT_BIT_STREAM_CTRL, data);
-		if (cpu_is_mx6q())
-			VpuWriteReg(BIT_FRAME_MEM_CTRL, IMAGE_ENDIAN | 1 << 12);
-		else
-			VpuWriteReg(BIT_FRAME_MEM_CTRL, IMAGE_ENDIAN);
+		VpuWriteReg(BIT_FRAME_MEM_CTRL, IMAGE_ENDIAN);
 		VpuWriteReg(BIT_INT_ENABLE, 1 << INT_BIT_PIC_RUN);
 		VpuWriteReg(BIT_AXI_SRAM_USE, 0);	/* init to not use SRAM */
 
@@ -2470,7 +2467,6 @@ RetCode vpu_DecOpen(DecHandle * pHandle, DecOpenParam * pop)
 
 		if (pDecInfo->mapType)
 			val |= (pDecInfo->tiledLinearEnable << 11 | 0x03 << 9);
-		val |= 1 << 12;
 	}
 	pCodecInst->ctxRegs[CTX_BIT_FRAME_MEM_CTRL] =
 		    val | (pDecInfo->openParam.chromaInterleave << 2);
-- 
1.8.0

